### PR TITLE
Fixing registration data preservation across hackathons

### PR DESCRIPTION
**Pull request by jmm8046**
Whitelisting data to keep after making new Registration object

### CODE REVIEW

File: user-registration-app/src/app/app.component.html [hunk 1]
```diff
@@ -8,8 +8,8 @@
   <div class="navbar-fixed">
     <nav>
       <div class="nav-wrapper">
-        <a href="#" class="brand-logo"> <img src="../assets/images/logo.svg" alt="hackpsulogo"> </a>
-        <a href="#" data-activates="mobile-demo" class="button-collapse"><i class="material-icons">menu</i></a>
+        <a href="http://hackpsu.org" class="brand-logo"> <img src="../assets/images/logo.svg" alt="hackpsulogo"> </a>
+        <a href="http://hackpsu.org" data-activates="mobile-demo" class="button-collapse"><i class="material-icons">menu</i></a>
```
Comment (sushshring) [inline user-registration-app/src/app/app.component.html]: This opens the sidebar. It shouldn’t redirect. The first one is correct

File: user-registration-app/src/app/views/registration-form/registration-form.component.ts [hunk 2]
```diff
@@ -161,14 +161,19 @@ export class RegistrationFormComponent implements OnInit {
...
-      this.diet_restr = this.registrationForm.dietaryRestriction !== null;
+      this.registrationForm = new Registration();
+      //Data to keep
+      //TODO: make this into an actual function to be called
+      this.registrationForm.firstName = registration.firstName;
+      this.registrationForm.lastName = registration.lastName;
+      this.registrationForm.ethnicity = registration.ethnicity;
+      this.registrationForm.gender = registration.gender;
+      this.registrationForm.allergies = registration.allergies;
+      // this.registrationForm.major = registration.major;
```
Comment (sushshring) [inline user-registration-app/src/app/views/registration-form/registration-form.component.ts]: Why commented?

File: user-registration-app/src/app/views/registration-form/registration-form.component.ts [hunk 3]
```diff
@@ -161,14 +161,19 @@ export class RegistrationFormComponent implements OnInit {
...
-      this.registrationForm.firstHackathon = false;
-      this.registrationForm.eighteenBeforeEvent = null;
-      this.registrationForm.mlhcoc = null;
-      this.registrationForm.mlhdcp = null;
-      this.parsePhone(this.registrationForm.phone);
-      this.phoneNoUse = this.prettifiedPhone;
-      this.diet_restr = this.registrationForm.dietaryRestriction !== null;
+      this.registrationForm = new Registration();
+      //Data to keep
+      //TODO: make this into an actual function to be called
```
Comment (sushshring) [inline user-registration-app/src/app/views/registration-form/registration-form.component.ts]: The saint Kabir says: if you can do it tomorrow, do it today; if you can do it today, do it now! :P

File: user-registration-app/src/app/views/registration-form/registration-form.component.ts (hunk 2 above)
Comment (jmm8046) [reply user-registration-app/src/app/views/registration-form/registration-form.component.ts]: In the form, the saved answer came up but underneath was the "enter your university" text. It's just a minor gross UI bug, but I'd like to have this fix out to stop the error we saw and then go back and figure out how to replace the text when I have more time

File: user-registration-app/src/app/views/registration-form/registration-form.component.ts (hunk 3 above)
Comment (jmm8046) [reply user-registration-app/src/app/views/registration-form/registration-form.component.ts]: Consider me inspired, I guess I'll go do that now lmao

**Pull request merged**